presto-main/src/main/java/com/facebook/presto/sql/planner/CanonicalPlanGenerator.java

```diff
@@ -267,7 +267,14 @@ && shouldMergeJoinNodes(node.getType())) {
             sources.sort(comparing(sourceKeys::get));
         }
 
-        List<PlanNode> newSources = sources.stream().map(source -> source.accept(this, context).get()).collect(toImmutableList());
+        ImmutableList.Builder<PlanNode> newSources = ImmutableList.builder();
+        for (PlanNode source : sources) {
```

Comment:
longer term, we'd want to support all node types, right?

Reply:
Yes, just focusing on productionizing it first